### ISSUE TITLE
Support forward to unix socket

### ISSUE THREAD

**Issue by tmds** [sshnet/SSH.NET#238]
OpenSSH supports forwarding to unix sockets: https://lwn.net/Articles/609321/.

For example, a mysql server on `remotehost` using socket `/var/lib/mysql/mysql.sock` can be made accessible on the local machine on TCP port `5000`:
```
ssh -L 127.0.0.1:5000:/var/lib/mysql/mysql.sock remotehost
```

**Comment by LightCC** [comment]
Are you just talking about a normal forwarded port using SSH?  This functionality is already built-in.  See an example here: https://stackoverflow.com/questions/41152737/c-sharp-ssh-net-tunnel-mysql-client-ioexception.

Check out ForwardedPort (base class), ForwardedPortLocal, ForwardedPortRemote, and ForwardedPortDynamic classes.

There are a few quirks though...

**Comment by drieseng** [comment]
@LightCC No, he's talking about [unix domain sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).

Don't hesitate to file issues for those "quirks".
